demos/coreHTTP/http_demo_s3_download_multithreaded.c

```diff
@@ -0,0 +1,965 @@
+
+/* Include demo config as the first non-system header. */
+#include "http_demo_s3_download_multithreaded_config.h"
+
+/* Include common demo header. */
+#include "aws_demo.h"
+
+/* Transport interface implementation include header for TLS. */
+#include "transport_secure_sockets.h"
+
+/* Common HTTP demo utilities. */
+#include "http_demo_utils.h"
+
+/* HTTP API header. */
+#include "core_http_client.h"
+
+/*------------- Demo configurations -------------------------*/
+
+/* Check that the TLS port of the server is defined. */
+#ifndef democonfigHTTPS_PORT
+    #error "Please define a democonfigHTTPS_PORT."
+#endif
+
+/* Check that the root CA certificate is defined. */
+#ifndef democonfigROOT_CA_PEM
+    #error "Please define a democonfigROOT_CA_PEM."
+#endif
+
+/* Check the the queue size is defined. */
+#ifndef democonfigQUEUE_SIZE
+    #error "Please define a QUEUE_SIZE."
+#endif
+
+/* Check that the pre-signed GET URL is defined. */
+#ifndef democonfigS3_PRESIGNED_GET_URL
+    #define democonfigS3_PRESIGNED_GET_URL    "GET-URL"
+#endif
+
+/* Check that a timeout for transport send and receive functions is defined.*/
+#ifndef democonfigTRANSPORT_SEND_RECV_TIMEOUT_MS
+    #define democonfigTRANSPORT_SEND_RECV_TIMEOUT_MS    ( 5000 )
+#endif
+
+/* Check that size of the user buffer is defined. */
+#ifndef democonfigUSER_BUFFER_LENGTH
+    #define democonfigUSER_BUFFER_LENGTH    ( 4096 )
+#endif
+
+/* Check that the range request length is defined. */
+#ifndef democonfigRANGE_REQUEST_LENGTH
+    #define democonfigRANGE_REQUEST_LENGTH    ( 2048 )
+#endif
+
+/**
+ * @brief The maximum number of times to run the loop in this demo.
+ */
+#ifndef httpexampleMAX_DEMO_COUNT
+    #define httpexampleMAX_DEMO_COUNT    ( 3 )
+#endif
+
+/**
+ * @brief Length of the pre-signed GET URL defined in http_demo_s3_download_multithreaded_config.h.
+ */
+#define httpexampleS3_PRESIGNED_GET_URL_LENGTH               ( sizeof( democonfigS3_PRESIGNED_GET_URL ) - 1 )
+
+/**
+ * @brief The length of the HTTP GET method.
+ */
+#define httpexampleHTTP_METHOD_GET_LENGTH                    ( sizeof( HTTP_METHOD_GET ) - 1 )
+
+/**
+ * @brief Field name of the HTTP range header to read from server response.
+ */
+#define httpexampleHTTP_CONTENT_RANGE_HEADER_FIELD           "Content-Range"
+
+/**
+ * @brief Length of the HTTP range header field.
+ */
+#define httpexampleHTTP_CONTENT_RANGE_HEADER_FIELD_LENGTH    ( sizeof( httpexampleHTTP_CONTENT_RANGE_HEADER_FIELD ) - 1 )
+
+/**
+ * @brief The HTTP status code returned for partial content.
+ */
+#define httpexampleHTTP_STATUS_CODE_PARTIAL_CONTENT          206
+
+
+/**
+ * @brief Time in ticks to wait between each iteration of
+ * RunCoreHttpS3DownloadMultithreadedDemo().
+ */
+#define httpexampleDELAY_BETWEEN_DEMO_ITERATIONS_TICKS    ( pdMS_TO_TICKS( 5000U ) )
+
+/**
+ * @brief The stack size to use for HTTP tasks.
+ */
+#define httpexampleTASK_STACK_SIZE                        ( configMINIMAL_STACK_SIZE * 2 )
```

Comment:
How was this stack size choosen? Could you see if this was a good stack size for all the platforms that we support?

Reply:
Was testing and accidentally checked that in -- thanks for catching. I updated it to be consistent with other similar demos.